priv/samples/basics/spi_rgb_leds.livemd

```diff
@@ -0,0 +1,371 @@
+# SPI RGB LEDs
+
+## Table of Contents
+1. [Prerequisites](#prerequisites)
+1. [Introduction](#introduction)
+1. [Wiring and discovering your SPI bus](#wiring-and-discovering-your-spi-bus)
+1. [Formatting our SPI payload](#formatting-our-spi-serial-data-structure)
+1. [Blinking a single LED](#blinking-a-single-led)
+1. [Managing all the LEDs on a strip](#managing-all-the-leds-on-a-strip)
+1. [Simple larson scanner](#simple-larson-scanner)
+1. [Simple rainbow](#simple-rainbow)
+1. [What next](#what-next)
+1. [Acronyms](#acronyms)
+
+## Prerequisites
+
+* A Raspberry Pi or BeagleBone board.
+* SK9822 or APA102 RGB LED strip.
+* (Recommended) A 5 volt 3 amp external power supply.
+* (Recommended) A general understanding of the [SPI bus][sparkfun.SPI].
+
+## Introduction
+
+In this exercise, we will be lighting up an RGB LED strip using the `circuits.SPI` library. 
+
+If you are unfamiliar with the SPI bus, please take a moment to review the [circuits.SPI][circuits.SPI] online documentation.
+It should help you build a general understanding on how data is transferred between a controller and peripheral.
+
+## Wiring and discovering your SPI bus
+
+First and foremost, we need to wire up the board and LED strip. 
+Using your boards documentation find the pins for MISO and MOSI.  
+
+<pre>
+ __________________       __________________ 
+|             SCLK |---->| SCLK             |
+|  SPI        MOSI |---->| MOSI        SPI  |
+| Master      MISO |<----| MISO       Slave |
+|             SS   |---->| SS               |
+ ------------------       ------------------ 
```

Comment:
I love that you mentioned this! I actually added `COPI` and `CIPO` in the Acronyms, but didn't want to confuse anyone since all docs use the future deprecated language still. I'll add a small blurb about it.